cache-config/t3c-apply/util/util.go

```diff
@@ -432,3 +433,81 @@ func Touch(fn string) error {
 	myfile.Close()
 	return nil
 }
+
+func UpdateMaxmind(cfg config.Cfg) bool {
+
+	if cfg.MaxMindLocation == "" {
+		return false
+	}
+
+	// Dont update for report mode
+	if cfg.RunMode == t3cutil.ModeReport {
+		return false
+	}
+
+	// Split url, get filename
+	url, err := url.Parse(cfg.MaxMindLocation)
+	if err != nil {
+		log.Errorf("error parsing maxmind url: %v", err)
+		return false
+	}
+	path := url.Path
+	pathParts := strings.Split(path, "/")
+	fileName := pathParts[len(pathParts)-1]
+
+	// Check if filename exists in ats etc
+	filePath := cfg.TsConfigDir + "/" + fileName
+	stdOut, _, code := t3cutil.Do(`date`,
+		"+%a, %d %b %Y %r %Z",
+		"-u",
+		"-r",
+		filePath)
+
+	if code == 0 {
+		// The file exists so run an IMS check
+		ims := fmt.Sprintf("If-Modified-Since: %s", string(stdOut[:len(stdOut)-1]))
+		stdOut, _, code := t3cutil.Do(`curl`,
+			"-I",
+			"-H",
+			ims,
+			"-sw",
+			"%{http_code}",
+			cfg.MaxMindLocation,
+			"-o",
+			"/dev/null")
+
+		if string(stdOut) == "304" {
+			log.Errorf("Received a 304 for maxmind database, not updating")
+			return false
+		}
+
+		if code != 0 {
+			log.Errorf("error issuing IMS for maxmind database")
+			return false
+		}
+	}
+
+	_, _, code = t3cutil.Do(`curl`,
+		"-so",
+		filePath,
+		cfg.MaxMindLocation)
+
+	if code != 0 {
+		log.Errorf("Error downloading maxmind database")
+		return false
+	}
+
+	gunzip := exec.Command("bash", "-c", "gunzip < "+filePath+" > "+strings.TrimSuffix(filePath, ".gz"))
+	err = gunzip.Run()
+	if err != nil {
+		log.Errorf("error running gunzip: %v\n", err)
+		return false
+	}
+	if code == 0 {
+		log.Errorf("Maxmind DB at %s successfully updated from %s", filePath, cfg.MaxMindLocation)
+		return true
+	}
+
+	log.Errorf("failure gunzip maxmind db")
```

Comment:
Yea I can just reorganize this area, i think some of it was leftover from attempts at using .Do, which seemed to have issues with using the redirection commands that were necessary and so exec.Command was the method I got working first.

Either way these blocks aren't necessary any more and the gunzip err if there is one gets logged inside the err check anyway and these other checks can be removed since they were already checked (and exited on failure if needed) earlier